pkg/kubelet/dockershim/helpers_test.go

```diff
@@ -271,16 +295,51 @@ func TestEnsureSandboxImageExists(t *testing.T) {
 			calls:       []string{"inspect_image"},
 			err:         true,
 		},
+		"should return error when image pull needs private auth, but none provided": {
+			injectImage:  true,
+			imgNeedsAuth: true,
+			injectErr:    libdocker.ImageNotFoundError{ID: "image_id"},
+			calls:        []string{"inspect_image", "pull"},
+			err:          true,
+		},
+		"should pull private image using dockerauth if image doesn't exist": {
+			injectImage:  true,
+			imgNeedsAuth: true,
+			injectErr:    libdocker.ImageNotFoundError{ID: "image_id"},
+			calls:        []string{"inspect_image", "pull"},
+			configJSON:   authJSON,
+			err:          false,
+		},
 	} {
+
 		t.Logf("TestCase: %q", desc)
 		_, fakeDocker, _ := newTestDockerService()
 		if test.injectImage {
-			fakeDocker.InjectImages([]dockertypes.Image{{ID: sandboxImage}})
+			images := []dockertypes.Image{{ID: sandboxImage}}
+			fakeDocker.InjectImages(images)
+			if test.imgNeedsAuth {
+				fakeDocker.MakeImagesPrivate(images, authConfig)
+			}
 		}
 		fakeDocker.InjectError("inspect_image", test.injectErr)
+
+		tmpdir := ""
+		if test.configJSON != "" {
+			if d, err := writeDockerConfig(test.configJSON); err != nil {
+				t.Errorf("could not create temp docker config file: %v", err)
+			} else {
+				dockerCfgSearchPath = append([]string{filepath.Join(d, ".docker")}, dockerCfgSearchPath...)
```

Comment:
If you want to monkey patch the variable, you'll need to defer a function to restore the original value. The code keeps appending to the same variable for each test case in the loop, making them order dependent. It also doesn't clean up after itself and pollute the environment for other tests.

You could still fall back to the thin wrapper if you'd like :-)